### ISSUE TITLE
Email click stats shows 0s for emails that have clicks

### ISSUE THREAD

**Issue by dzver** [Automattic/wp-calypso#97638]
### Quick summary

The email clicks consistently shows 0% even for emails that have a non-trivial number of clicks and are clearly not at 0%.

### Steps to reproduce

Visit the stats for veselin.blog
1. Visit https://wordpress.com/stats/subscribers/veselin.blog - all emails have 0% CTR
2. Check the individual emails - they actually have clicks. Having 7 clicks with 129 recepients and 42 opens doesn't result into a 0% CTR 
3. https://wordpress.com/stats/email/clicks/day/5558/veselin.blog this email shows as 0.8% in the summary and 1% in the details. 7/129 is 4.6%, and 7/30 is 23.3% - it's simply never 0%, 0.8%, or 1%, no matter how I calculate it.

<p float="left">

<img width="20%" alt="Image" src="https://github.com/user-attachments/assets/aae4a0b0-77a9-44c6-b3e2-4b4c87fc72b0" />
<img width="20%" alt="Image" src="https://github.com/user-attachments/assets/a261f828-57e5-4a09-92ea-59d16053c94e" />
<img width="20%" alt="Image" src="https://github.com/user-attachments/assets/01006757-9cef-45a3-9ea3-51f3ab03c294" />

</p>

### What you expected to happen

Email stats to show actual numbers, exact clicks rather than %s

### What actually happened

0s and wrong %s

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### If the above answer is "Yes...", outline the workaround.

Flagging that some users are impacted because this only affects users with email subscribers.

**Comment by Robertght** [comment]
I'm able to replicate this with the latest couple of posts from my self-hosted environment too.

@Automattic/red @Automattic/jetpack-bebop is this something one of you can look into?

**Comment by a8ck3n** [comment]
> View email clicks rate by post. Returns mock data.

Hmm…maybe I'm looking in the wrong spot. Either way, will give it another look next week.

**Comment by a8ck3n** [comment]
Looped in Zap before moving forward with this one.

p1734618654975259-slack-C052XEUUBL4

**Comment by lezama** [comment]
https://github.com/Automattic/wp-calypso/pull/97729 adds two decimal places and tooltips to the rates for better clarity.

One potential issue is that, for posts published before August 2023, we don’t have unique counts. Since rates are calculated using unique counts, we need to consider how to display rates for those posts.

**Comment by lezama** [comment]
**TL;DR: Posts without links in the content don’t have unique clicks recorded, so the click rate is shown as 0 for these.**

I’ve been investigating the email clicks issue, and here are my findings so far, along with some points that may not be clear to users:

Previously, we displayed Total clicks in the stats. Now, we calculate and display **Click rate**, which are **derived from unique clicks**.

Data Storage

* Unique clicks are stored in the email_link_stats table. However, this table is designed to record clicks for links in the email content, not for clicks on the post title.

* Total clicks, on the other hand, are leveraged from the `wpcom_email_click` tracks event. Then stored in the email_stats_daily_ table after being parsed from logs.

To make it even more complicated, we record `wpcom_email_click` events in different parts of the code depending if it's a click in the post title or not.

For clicks on post titles in emails, this is the backtrace for when this event is triggered:

```
[0] => wp-content/lib/tracks/class.wpcom-tracks-client.php:62
[1] => wp-content/lib/tracks/client.php:32
[2] => wp-content/mu-plugins/groovemails.php:282
```

For clicks on non-post-title links we also record wpcom_email_click, but with a different backtrace:

```
[0] => wp-content/lib/tracks/class.wpcom-tracks-client.php:62
[1] => wp-content/lib/class.wpcom-user-content-links.php:292

```

These discrepancies in how clicks are tracked and stored likely contribute to the confusion and inconsistencies in the reported stats. cc @Automattic/zap @crisbusquets

**Comment by simison** [comment]
Either someone in Creators (cc @lessbloat) or Jetpack Stats (cc @ederrengifo).

**Comment by escapemanuele** [comment]
Hey @davemart-in, I moved this to the Loop board, let me know if we need to move it elsewhere.

**Comment by davemart-in** [comment]
This feels more like a stats infrastructure thing. @Automattic/red does this feel like a good fit for your team to pick up?

**Comment by dzver** [comment]
@davemart-in, from what I understand from[ the comments here](https://github.com/Automattic/wp-calypso/issues/97638#issuecomment-2567182492) and on the other similar issues:

- The endpoints return two numbers. One is the number of clicks anywhere, and the other the number of clicks on in-post links. - Our language is misleading, naming the in-post links as unique
- We should change that language
- We should only use the other number of clicks anywhere in stats
- We can stop showing relative numbers and switch to absolute numbers for best results

**Comment by simison** [comment]
Absolute numbers aren't really helpful as-is because sending/open rate fluctuates, and % is what other platforms also use. 

For the rest, yep, we shouldn't have two ways to track clicks. Clicking author/logo/titles/sitename should be as important as links in content, but clicks on "manage subscription", "apps" etc aren't really relevant to publisher.
